addons/survey/views/survey_templates.xml

```diff
@@ -30,7 +30,7 @@
                             <t t-set="page_number" t-value="page_ids.index(page.id) + (1 if survey.progression_mode == 'number' else 0)"/>
                         </t>
                         <t t-else="">
-                            <t t-set="page_ids" t-value="survey.question_ids.ids"/>
```

Comment:
Either "predefined_question_ids" is always set, and then you don't have to test it in the python, either it's not and then you have to test the selection_mode here as well.

I think it should always be set considering the "_prepare_user_input_predefined_questions" method, but should be double-checked.
Would allow simplifying the line above and avoid the continuation with the backslash.

Reply:
In 14.0 it looks like if I let only `predefined_question_ids` this works with no problem.
Unfortunately, this is not the case in 16.0, if you enter a live session it doesn't take into account the random selection of questions.

That's why I made it this way to prevent the future crashes in 16.0, so I'll also add the check in the xml template